contributors/design-proposals/grow-volume-size.md

```diff
@@ -0,0 +1,301 @@
+| ----------------| :---------------: | :--------------------------:| :----------------------: |
+| EBS             | Yes               | Yes                         | Yes                      |
+| GCE PD          | Yes               | Yes                         | Yes                      |
+| GlusterFS       | Yes               | No                          | Yes                      |
+| Cinder          | Yes               | Yes                         | Yes                      |
+| Vsphere         | Yes               | Yes                         | No                       |
+| Ceph RBD        | Yes               | Yes                         | No                       |
+| Host Path       | No                | No                          | No                       |
+| Azure Disk      | Yes               | Yes                         | No                       |
+| Azure File      | No                | No                          | No                       |
+| Cephfs          | No                | No                          | No                       |
+| NFS             | No                | No                          | No                       |
+| Flex            | Yes               | Maybe                       | No                       |
+| LocalStorage    | Yes               | Yes                         | No                       |
+| Block device    | Yes               | No                          | No                       |
+
+
+## Implementation Design
+
+For volume type that requires both file system expansion and a volume plugin based modification, growing persistent volumes will be two
+step process.
+
+
+For volume types that only require volume plugin based api call, this will be one step process.
+
+### Prerequisite
+
+* `pvc.spec.resources.requests.storage` field of pvc object will become mutable after this change.
+* #sig-api-machinery has agreed to allow pvc's status update from kubelet as long as pvc and node relationship
+  can be validated by node authorizer.
+* This feature will be protected by an alpha feature gate, so as API changes needed for it.
+
+### Admission Control and Validations
+
+* Resource quota code has to be updated to take into account PVC expand feature.
+* In case volume plugin doesn’t support resize feature. The resize API request will be rejected and PVC object will not be saved. This check will be performed via an admission controller plugin.
+* In case requested size is smaller than current size of PVC. A validation will be used to reject the API request. (This could be moved to admission controller plugin too.)
+* Not all PVCs will be resizable even if underlying volume plugin allows that. Only dynamically provisioned volumes
+which are explicitly enabled by an admin will be allowed to be resized. A plugin in admission controller will forbid
+size update for PVCs for which resizing is not enabled by the admin.
+
```

Comment:
@humblec no we don't.  Can you explain what scenario you had in mind that would cause problem if user deletes PVC while resize is in progress?